src/controllers/mainController.ts

```diff
@@ -223,6 +224,18 @@ export default class MainController implements vscode.Disposable {
 			const providerInstance = new this.ExecutionPlanCustomEditorProvider(this._context, this.executionPlanService, this._untitledSqlDocumentService);
 			vscode.window.registerCustomEditorProvider('mssql.executionPlanView', providerInstance);
 
+			if (this.isExperimentalEnabled) {
```

Comment:
NIT: Can you put the if condition inside the URI handler and always register this handler?